nntrainer/layers/bn_layer.cpp

```diff
@@ -198,28 +187,15 @@ void BatchNormalizationLayer::calcDerivative(RunLayerContext &context) {
   Tensor &dx = context.getOutgoingDerivative(SINGLE_INOUT_IDX);
   Tensor &deviation = context.getTensor(wt_idx[BNParams::deviation]);
   Tensor &invstd = context.getTensor(wt_idx[BNParams::invstd]);
+  Tensor &cvar = context.getTensor(wt_idx[BNParams::cvar]);
 
   Tensor &t_reduced = context.getTensor(wt_idx[BNParams::t_reduced]);
-  Tensor &t_full = context.getTensor(wt_idx[BNParams::t_full_bw]);
-  Tensor cvar = t_reduced;
-
-  if (context.getTrainable()) {
-    /**
-     * This implementation depends on the pre-calculated dbeta calculated.
-     */
-    Tensor &dbeta = context.getWeightGrad(wt_idx[BNParams::beta]);
-    t_reduced = dbeta.divide(divider);
```

Comment:
this is a bus in original implementation. This assumes that the t_reduced is not needed anymore, but it is as it shared memory with c_var. This is working for now because this statement creates new memory.